packages/web3-core/src/web3_subscription_manager.ts

```diff
@@ -1,16 +1,20 @@
-import { ProviderError, SubscriptionError } from 'web3-common';
+import { ProviderError, SubscriptionError, Web3APISpec } from 'web3-common';
 import { isSupportSubscriptions } from './utils';
 import { Web3RequestManager, Web3RequestManagerEvent } from './web3_request_manager';
-import { Web3Subscription, Web3SubscriptionConstructor } from './web3_subscriptions';
+import { Web3SubscriptionConstructor } from './web3_subscriptions';
 
 export class Web3SubscriptionManager<
-	ST extends { [key: string]: Web3SubscriptionConstructor<Web3Subscription> },
+	API extends Web3APISpec,
+	RegisteredSubs extends { [key: string]: Web3SubscriptionConstructor<API> },
 > {
-	private readonly _subscriptions: Map<string, Web3Subscription> = new Map();
+	private readonly _subscriptions: Map<
+		string,
+		InstanceType<RegisteredSubs[keyof RegisteredSubs]>
+	> = new Map();
 
 	public constructor(
-		public readonly requestManager: Web3RequestManager,
-		public readonly registeredSubscriptions: ST,
+		public readonly requestManager: Web3RequestManager<API>,
+		public readonly registeredSubscriptions: RegisteredSubs,
```

Comment:
We should stay consistent and not abbreviate if not needed

Reply:
Yes, it was old code. in refactored I removed the abbreviation.